src/iter.rs

```diff
@@ -85,3 +86,158 @@ fn hex_chars_to_byte(hi: u8, lo: u8) -> Result<u8, HexToBytesError> {
     let ret = (hih << 4) + loh;
     Ok(ret as u8)
 }
+
+/// Iterator over bytes which encodes the bytes and yields hex characters.
+pub struct BytesToHexIter<I: ExactSizeIterator + FusedIterator + Iterator<Item = u8>> {
+    /// The iterator whose next byte will be encoded to yield hex characters.
+    iter: I,
+    /// The low character of the pair (high, low) of hex characters encoded per byte.
+    low: Option<char>,
+}
+
+impl<I> BytesToHexIter<I>
+where
+    I: ExactSizeIterator + FusedIterator + Iterator<Item = u8>,
+{
+    /// Constructs a new `BytesToHexIter` from a byte iterator.
+    pub fn new(iter: I) -> BytesToHexIter<I> { Self { iter, low: None } }
+}
+
+impl<I> Iterator for BytesToHexIter<I>
+where
+    I: ExactSizeIterator + FusedIterator + Iterator<Item = u8>,
```

Comment:
This shouldn't require `ExactSizeIterator`.

Reply:
Thanks, fixed.